src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -672,6 +678,52 @@ def ok_to_stop(self,
             return HandleCommandResult(-errno.EBUSY, '', warn_message)
         return HandleCommandResult(0, warn_message, '')
 
+    def find_prometheus_credentials(self, multicluster_config: Dict[str, Any], url: str) -> Optional[Dict[str, Any]]:
+        for _, clusters in multicluster_config['config'].items():
```

Comment:
We need this here because we store the remote cluster's prometheus credentials in the multi_cluster config. So when we populate the configs in the prometheus.yml.j2 from here, we send these credentials along with other configs to set it in the prometheus.yml.

Reply:
CC: @adk3798 @rkachach are you guys ok with Cephadm being multi-cluster aware?

I personally think this goes beyond the scope of Cephadm and multi-cluster feature, which is/was a Ceph Dashboard only thing. If we wanted/needed to involve Cephadm/Orch, then probably smarter choices could have been taken regarding the implementation.

@aaSharma14 can't this logic be put in the Dashboard side? I don't find any reason good enough why we couldn't. It might need more calls, but we shouldn't create new interfaces out of convenience if there are already some that provide what's needed. That leads to an ad-hoc API (vs generic one), which is unsustainable.

Reply:
@epuertat , moved the logic to dashboard side, PTAL